src/ISBootloaderISB.cpp

```diff
@@ -992,13 +1013,17 @@ is_operation_result cISBootloaderISB::download_image(std::string filename)
 
     status_update("(ISB) Erasing flash...", IS_LOG_LEVEL_INFO);
 
-    if(erase_flash() != IS_OP_OK) return IS_OP_ERROR;
-    if(select_page(0) != IS_OP_OK) return IS_OP_ERROR;
+    result = erase_flash();
+    CHECK_IS_OP_DWNLD()
```

Comment:
I'm not sure I see the value in this macro.
I think just updating the existing condition check to  `if ((result = erase_flash()) != IS_OP_OK) { fclose(firmware_file); return result; }` is more readable, than having to move back up the code the find what CHECK_IS_OP_DWNLD() is doing.  Plus, its clearer when stepping through with the debugger.

Alternatively, in the lower calls to CHECK_IS_OP_DWNLD(), you could still keep the `result = function_op()` and just add the `if (result != IS_OP_OK) { ... }.

Finally, it might get a little "nesty", but also doing this is a nested fashion, might also be appropriate:
`if ( (result=erase_flash()) == IS_OP_OK) then { if (result=select_page()) == IS_OP_OK) { ... } } fclose(firmware_file);` then in reading the nested code, it is straight forward that failure of one, means non-execution of the nested, and we keep a consistent, singular close/return result).

Just suggestions though... as long as it works!